server/src/main/java/org/opensearch/index/seqno/LocalCheckpointTracker.java

```diff
@@ -131,22 +131,49 @@ public synchronized void markSeqNoAsProcessed(final long seqNo) {
      * @param seqNo the sequence number to mark as persisted
      */
     public synchronized void markSeqNoAsPersisted(final long seqNo) {
-        markSeqNo(seqNo, persistedCheckpoint, persistedSeqNo);
+        markSeqNo(seqNo, persistedCheckpoint, persistedSeqNo, false);
     }
 
-    private void markSeqNo(final long seqNo, final AtomicLong checkPoint, final LongObjectHashMap<CountedBitSet> bitSetMap) {
+    /**
+     * Updates the processed checkpoint to the provided sequence number if segment replication is enabled
+     *
+     * @param seqNo the sequence number to mark as processed
+     */
+    public synchronized void fastForwardProcessedSeqNo(final long seqNo) {
+        markSeqNo(seqNo, processedCheckpoint, null, true);
+    }
+
+    private boolean shouldUpdateSeqNo(final long seqNo, boolean segrep, final AtomicLong checkpoint) {
+        return !((seqNo <= checkpoint.get()) || (segrep && seqNo > persistedCheckpoint.get()));
```

Comment:
I think you can put the check against the persisted checkpoint inside of `fastForwardProcessedSeqNo` and avoid the segrep specific flag.

Reply:
Changed the logic in the check to avoid segrep specific flag